### ISSUE TITLE
try/except and no-else-return

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#1852]
```
def function(condition):
    if condition:
        return True
    else:
        try:
            return False
        except ValueError:
            return None
```

This code should get flagged with `Unnecessary "else" after "return" (no-else-return)`, but it doesn't.

### pylint --version output
```
pylint 2.0.0
astroid 1.6.0
Python 3.6.4 (default, Dec 25 2017, 14:57:46) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

**Comment by PCManticore** [comment]
Thanks, sounds reasonable!
